arviz/data/inference_data.py

```diff
@@ -328,16 +328,30 @@ def from_netcdf(filename: str) -> "InferenceData":
         InferenceData object
         """
         groups = {}
-        with nc.Dataset(filename, mode="r") as data:
-            data_groups = list(data.groups)
+        try:
+            with nc.Dataset(filename, mode="r") as data:
+                data_groups = list(data.groups)
 
-        for group in data_groups:
-            with xr.open_dataset(filename, group=group) as data:
-                if rcParams["data.load"] == "eager":
-                    groups[group] = data.load()
-                else:
-                    groups[group] = data
-        return InferenceData(**groups)
+            for group in data_groups:
+                with xr.open_dataset(filename, group=group) as data:
+                    if rcParams["data.load"] == "eager":
+                        groups[group] = data.load()
+                    else:
+                        groups[group] = data
+            return InferenceData(**groups)
+        except OSError as e:  # pylint: disable=invalid-name
```

Comment:
To use `e` (instead of `err` or something a bit longer that pylint considers optimal) as variable to raise errors from other errors, it may be worth it to add it to `.pylintrc` instead of having to disable the error every time